battle-service/src/main/java/com/company/battle/controllers/GameController.java

```diff
@@ -1,43 +1,74 @@
 package com.company.battle.controllers;
 
+import com.company.battle.mappers.GameMapper;
+import com.company.battle.mappers.SettingsMapper;
+import com.company.battle.services.GameService;
 import com.company.common.dtos.GameDto;
-import com.company.common.dtos.SettingsDto;
+import com.company.common.dtos.SearchGameRequestDto;
+import com.company.common.models.GameEntity;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 @RestController
 @RequestMapping("api/v1/game")
+@RequiredArgsConstructor
 public class GameController {
 
+    @Autowired
+    private final GameService gameService;
+
+    private final GameMapper gameMapper = GameMapper.INSTANCE;
+
+    private final SettingsMapper settingsMapper = SettingsMapper.INSTANCE;
+
     @GetMapping("/{id}")
     public GameDto getGameById(@PathVariable UUID id) {
-        return null;
+        GameEntity gameEntity = gameService.getById(id);
+        return gameMapper.toDTO(gameEntity);
     }
 
-    @GetMapping("/{link}")
-    public GameDto getGameByLink(@PathVariable String link) {
-        return null;
+    @GetMapping("/public-games")
+    public List<GameDto> getPublicGames() {
+        List<GameEntity> gameEntities = gameService.getPublicGames();
+        return gameEntities.stream()
+                .map(gameMapper::toDTO)
+                .collect(Collectors.toList());
     }
 
-    @GetMapping
-    public List<GameDto> getActiveGames() {
-        return null;
+    @GetMapping("/search")
+    public List<GameDto> searchGames(@Valid @RequestBody SearchGameRequestDto requestDto) {
+        List<GameEntity> gameEntities = gameService.findGame(requestDto);
```

Comment:
i would introduce GameSearchFacade which will accept SearchGameRequestDto and construct CriteriaBuilder, after that it will call 
GameService.findbGame(CriteriaBuilder searchCriteria)

Reply:
I added specifications. Do I no longer need to make a GameSearchFacade?